rust-runtime/aws-smithy-runtime-api/src/client/interceptors/error.rs

```diff
@@ -169,3 +169,31 @@ impl std::error::Error for InterceptorError {
         self.source.as_ref().map(|err| err.as_ref() as _)
     }
 }
+
+/// A convenience error that allows for adding additional `context` to `source`
+#[derive(Debug)]
+pub struct ContextAttachedError {
+    context: String,
+    source: Option<BoxError>,
+}
+
+impl ContextAttachedError {
+    pub fn new(context: impl Into<String>, source: BoxError) -> Self {
```

Comment:
You can make source `impl Into<BoxError>` to simplify calling `new`.

Reply:
Why not 👍  Addressed in 93d044d.